flink/interpreter/src/main/scala/org/apache/zeppelin/flink/FlinkScalaInterpreter.scala

```diff
@@ -829,6 +835,12 @@ class FlinkScalaInterpreter(val properties: Properties) {
       }
     }
   }
+
+  private def replaceYarnAddress(webURL: String, yarnAddress: String): String = {
+    val pattern = "(http://.*:\\d+)(/.*)".r
```

Comment:
I maybe would change it to the to parsing of `webURL` into URL or URI, and construction of the new object, but with changed components. 
Also it won't handle `https` schema, etc.

Reply:
@alexott  `URL` doesn't expose set method. So it is not easy to construct URL with components, however I fixed the https issue and also add unit tests